### ISSUE TITLE
[Feature request] Integration With SponsorBlock API

### ISSUE THREAD

**Issue by Abystus** [iv-org/invidious#3130]
**Is your feature request related to a problem? Please describe.**
Not being able to skip the following types of sponsor ads/promotions:

1. Sponsors
2. Intermission/Intro Animation
3. Endcards/Credits
4. Preview/Recap
5. Interaction Reminder (Subscribe)
6. Unpaid/Self Promotion
7. Music: Non-Music Section
8. Highlight
9. Filler Tangent

**Describe the solution you'd like**
Utilize the SponsorBlock API from [sponsor.ajay.app](https://sponsor.ajay.app/)

The above types can be checked off similar to how Piped does it in their preferences:

![image](https://user-images.githubusercontent.com/1529049/170921473-1d811a48-b712-4096-9c28-f75a855f133c.png)

Basically when a sponsor ad is detected, the video skips ahead until the end of the ad and continues the video. It will do this for all types toggled in the preferences and works extremely well.

**Describe alternatives you've considered**
Piped has these options, and I normally use them for advert heavy channels.

**Comment by unixfox** [comment]
Use the official extension: https://github.com/ajayyy/SponsorBlock#sponsorblock

It support invidious.
